File: src/types/orbit.ts
export type OrbitSummary = {
  shareId: string;
  summary: string;
  keyPoints?: string[];
  createdAt: string;
  updatedAt: string;
};

export type UserInterests = {
  interests: string[];
  updatedAt: string;
};

export type OrbitLearningLesson = {
  title: string;
  day: number;
  totalDays: number;
  overview: string;
  paragraphs: string[];
  code?: string[]; // Optional preformatted snippets (e.g., source code)
  quiz: Array<{
    question: string;
    answers: string[];
    correctAnswer: string;
  }>;
};

export type OrbitLearningPlan = {
  id: string;
  topic: string;
  depth: "light" | "standard" | "deep" | "expert" | "auto";
  totalLessons: number;
  currentLesson: number;
  startedAt: string;
  syllabus?: Array<{
    day: number;
    title: string;
    summary: string;
    quizFocus?: string;
  }>;
  completedLessons: Array<{
    day: number;
    title: string;
    overview: string;
  }>;
  updatedAt: string;
  lastLessonGeneratedAt?: string;
  activeLesson?: OrbitLearningLesson;
  lastLessonDate?: string; // YYYY-MM-DD
};

export type OrbitLearningProfile = {
  plans: OrbitLearningPlan[];
  updatedAt: string;
};

export type DailySummaryPayload = {
  overview: string[];
  recommendations: string[];
  completedWork: WorkTaskHighlight[];
  pendingWork: WorkTaskHighlight[];
  insights: OrbitInsightCard[];
  learningLesson?: OrbitLearningLesson | null; // Deprecated, keep for backward compat
  learningLessons?: OrbitLearningLesson[];
};

export type OrbitLesson = OrbitLearningLesson & {
  id: string;
  topic: string;
  createdAt: string;
};

export type DailySummary = {
  shareId: string;
  date: string; // YYYY-MM-DD format
  shownAt: string; // ISO timestamp when shown to user
  createdAt: string;
  payload?: DailySummaryPayload | null;
};

export type OrbitInsightType = "news" | "concept";

export type OrbitInsightCard = {
  id: string;
  topic: string;
  title: string;
  summary: string;
  paragraphs: string[];
  type: OrbitInsightType;
  referenceUrl?: string | null;
};

export type OrbitInsightPreferences = {
  moreTopics: Record<string, number>;
  lessTopics: Record<string, number>;
  updatedAt: string;
};

export type InsightVoteDirection = "more" | "less";

export type WorkTaskHighlight = {
  title: string;
  status: string;
  note?: string | null;
};

File: android/toodl-share/app/src/main/java/com/toodl/share/data/model/Orbit.kt
package com.toodl.share.data.model

data class SharedLink(
    val id: String = "",
    val url: String? = null,
    val title: String? = null,
    val description: String? = null,
    val sourceApp: String? = null,
    val platform: String = "android-share",
    val contentType: String = "unknown", // link, video, article, etc.
    val tags: List<String> = emptyList(),
    val previewImageUrl: String? = null,
    val storagePath: String? = null,
    val status: String = "new", // new, saved, archived
    val summarizable: Boolean = false,
    val createdAt: String = "",
    val updatedAt: String = ""
)

data class DailySummary(
    val shareId: String = "",
    val date: String = "",
    val shownAt: String = "",
    val createdAt: String = "",
    val payload: DailySummaryPayload? = null
)

data class DailySummaryPayload(
    val overview: List<String> = emptyList(),
    val recommendations: List<String> = emptyList(),
    val completedWork: List<WorkTaskHighlight> = emptyList(),
    val pendingWork: List<WorkTaskHighlight> = emptyList(),
    val insights: List<OrbitInsightCard> = emptyList()
)

data class WorkTaskHighlight(
    val title: String = "",
    val status: String = "",
    val note: String? = null
)

data class OrbitInsightCard(
    val id: String = "",
    val topic: String = "",
    val title: String = "",
    val summary: String = "",
    val paragraphs: List<String> = emptyList(),
    val type: String = "concept",
    val referenceUrl: String? = null
)
